.github/workflows/ci-build-and-smoke.yml

```
@@ -0,0 +1,45 @@
+    branches: [ main ]
+
+jobs:
+  build:
+    name: Build and Test (dotnet + rust)
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v3
```

actionlint (1.7.7)

the runner of "actions/setup-dotnet@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)